### ISSUE TITLE
RTNETLINK answers: No such device

### ISSUE THREAD

**Issue by AinzOoalGown-fun** [FluxionNetwork/fluxion#1095]
Hello, I have this error ( RTNETLINK answers: No such device) when  i'm Starting Captive Portal access point service.

I waited but nothing happens after the error 
airmon-ng 

Can you help me please

PHY     Interface       Driver          Chipset

phy0    wlan0           rt2800usb       Ralink Technology, Corp. RT2870/RT3070    (injection and monitor work)
phy1    wlan1           rt2800usb       D-Link System AirPlus G DWL-G122(rev.E1) [Ralink RT2070] ( (injection and monitor no work)

![error](https://user-images.githubusercontent.com/73220716/175174866-06f36377-3747-4396-bb97-26805439ae71.jpg)

**Comment by cjb900** [comment]
This is not any error I've seen before. The only thing I can suggest would be to reboot the whole computer. unplug and replug in your wifi adapter's and before starting fluxion run ``sudo airmon-ng check kill`` and see if that helps.

**Comment by AinzOoalGown-fun** [comment]
Thank you it works, but it's not top fluxion, I tried it on my wifi, the password doesn't show and the login page bugs.

**Comment by cjb900** [comment]
I have a wifi adapter with the same RT2870/RT3070 chipset and fluxion works with no issues and the captive portal page is not buggy and the password shows. So would need more information on what you mean by "password doesn't show and the login page bugs"

**Comment by AinzOoalGown-fun** [comment]
I'm going to a small video

**Comment by AinzOoalGown-fun** [comment]
There are 2 videos, one on the PC and the other on the phone.
https://we.tl/t-9VUAvlGnq8 
Thank you -_-

**Comment by cjb900** [comment]
Well for the deauthing part not working there is a reason fluxion has more than one deauthign method. So if one doesn't work (airplay-ng) try one of the other options. I personally use MDK4 as it's more reliable than airplay-ng in my tests anyways. Another thing I noticed is that you are not using the latest or official version of fluxion. On the "Select a password verification method" you have pyrit listed as an option. Fluxion dropped support a few years ago because it was no longer available in the kali repositories and not needed for fluxion to work properly. So I would redownload the latest fluxion and see if that fixes any of your issues. I also tried to replicate the error (RTNETLINK answers: No such device) you are getting with my RT3070 and was unable to. So no idea where that error is coming from if it's something in your setup or something else in kali. 

The error you get with trying to connect to the fake AP and chrome not getting the captive portal is as expected. google.com uses https://. to get the captive portal if it does not automatically pop up you have to go to an http:// site, not an https:// one. there is no fix for this as it has to do with most https:// using HSTS these days. 

Regarding your phone not being able to connect to the fake AP I don't see anything obviously wrong with the 6 windows in fluxion and again testing fluxion on my end I can connect to the fake AP with no issues on both of my phones (Samsung S22 Ultra and iPhone 11). It's possible it's something with your phone but no way to tell from your video recording. 

Another thing that could be part of your issue is running wifi hacking attacks in a VM (VMware). Some here have had issues depending on their hardware setup when using a VM but running on hardware either dual-booting or from a live USB have no issues. 

So what I would do is delete whatever version of fluxion you have and download the latest one from this GitHub page. Than see if that version helps any. Also any time you are going to use fluxion run ``sudo airmon-ng check kill`` first. Regarding your deauthing issues, I would try each deauthing method in fluxion and see which one works best for your setup. If you still have issues I would stop using VMware and try running kali on hardware (dual-booting or live USB) and see if that helps any. 

If none of this helps since I can't replicate any of your issues it's hard to say what is causing these and would have to be something somewhere in your setup or with your installation of kali. 

Hope this helps.

**Comment by AinzOoalGown-fun** [comment]
thank you I will try

**Comment by AinzOoalGown-fun** [comment]
I can't find the official Fluxion GitHub
you can give it to me ?
Thank you

**Comment by cjb900** [comment]
You are already on it as you are communicating here which is the official fluxion github page.

**Comment by AinzOoalGown-fun** [comment]
https://github.com/FluxionNetwork/fluxion it  say Fluxion is a remake

**Comment by cjb900** [comment]
It's a remake of linset but this is the official fluxion github page

**Comment by AinzOoalGown-fun** [comment]
Why do you say that I don't have the official one, I have downloaded it since 
https://github.com/FluxionNetwork/fluxion 😂

**Comment by cjb900** [comment]
Because as I mentioned in your video under "Select a password verification method" you have pyrit listed as an option. Fluxion dropped support for pyrit a few years ago because it was no longer available in the kali repositories and not needed for fluxion to work properly. This here tells me that you either have an old version or downloaded it from another githup page.

**Comment by AinzOoalGown-fun** [comment]
HI -_- 

I have downloaded it since from:
https://github.com/FluxionNetwork/fluxion 
Pyrit is always in the last version fluxion .
Try download the latest version.

When I make ( airmon-ng check kill ) it always kills the same processes.
Killing these processes:
PID Name
1522 wpa_supplicant

I tried the MDK4 and it did not disconnect my devices. 
With airplay-ng it disconnects my devices  

Is it possible to bypass HSTS
https://github.com/FluxionNetwork/fluxion/issues/697


I think the problem comes from monitor mode wifi adapt which a Chinese copy.
the bug must come from this, the injection and the monitor work but its bug.

Can you give me the latest and greatest monitor mode wifi adapt that supports the injection and the monitor. 
Thank you -_-

**Comment by cjb900** [comment]
I don't know how you are downloading or where you are getting fluxion from but as you can see in my video [here](https://we.tl/t-q6mXQaOUuc) I downloaded a fresh copy and when I get to the "Select a password verification method" you can see there is no option for pyrit. So if the version you have has pyrit in it it could be part of your issues since pyrit was removed from fluxion a few years ago. 

Regarding running ``sudo airmon-ng check kill`` it's supposed to kill those 2 processes. Because those 2 processes can cause issues with wifi hacking. This is why running that command is recommended before starting any wifi hacking attacks.  

Regarding bypassing HSTS this is not doable these days with the security implementation that are in modern browsers. Even if it is somehow possible I'm not a developer of fluxion. I only help others with issues with it. So it would be up to the developers to look into this and if they find a way implement it in fluxion. 

Regarding your wifi adapter. I have a Chinese knock-off adapter with the same chipset (as you can see in the video I shared) and have not been able to replicate any of your issues in any of my tests. but below is a list of supported wifi adapters and chipsets if you want to look into getting a new one. 

Also if you have not tried running kali on hardware and not in a VM (VMware) it would be recommended to try that after getting the latest version of fluxion if you still have issues. 

Atheros AR9271 <- present in TP-Link TL-WN722N v1 or Alfa AWUS036NHA (2.4Ghz - USB)
MediaTek MT7610U <- present in TP-Link Archer T2UH or AWUS 036ACHM(2.4Ghz/5Ghz - USB)
MediaTek MT7612U <- present in Alfa AWUS036ACM (2.4Ghz/5Ghz - USB) (not recommended to be used on Virtual Machines, but it works very well on native Linux. The recommended Linux kernel version is 5.3 or greater). For Rpi 2/3 run this command to get the card working echo "options mt76_usb disable_usb_sg=1" > /etc/modprobe.d/mt76_usb.conf
Ralink RT2770 <- present in Alfa AWUS051NH (2.4Ghz/5Ghz - USB)
Ralink RT3070 <- present in TP-Link TL-WN7200ND or Alfa AWUS036NH (2.4Ghz - USB)
Ralink RT3572 <- present in Alfa AWUS052NHS (2.4Ghz/5Ghz - USB)
Ralink RT5372 <- present in D-Link DWA-137 (2.4Ghz - USB)
Ralink RT5378 <- present in some some unbranded cheap chinese dongles (2.4Ghz - USB)
Ralink RT5572 <- present in Panda PAU07 or Panda PAU09 (2.4Ghz/5Ghz - USB)
